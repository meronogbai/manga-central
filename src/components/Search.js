import { useState } from 'react';
import { useHistory } from 'react-router-dom';

const Search = () => {
  const [manga, setManga] = useState('');
  const history = useHistory();
  const handleChange = e => {
    setManga(e.target.value);
  };
  const handleSubmit = e => {
    e.preventDefault();
    history.push({
      pathname: '/manga',
      search: `?q=${manga.replace(' ', '%20')}`,
    });
  };
  return (
    <main>
      <form onSubmit={handleSubmit}>
        <input type="text" value={manga} onChange={handleChange} placeholder="Manga" required pattern="\S.*" />
        <button type="submit">Search</button>
      </form>
    </main>
  );
};

export default Search;
