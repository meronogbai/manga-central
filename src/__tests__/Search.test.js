import { create } from 'react-test-renderer';
import Search from '../components/Search';

test('renders correctly', () => {
  const tree = create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});
