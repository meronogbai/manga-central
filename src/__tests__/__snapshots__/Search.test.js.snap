// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="search"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      pattern="\\\\S.*"
      placeholder="Manga"
      required={true}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</main>
`;
